src/flows/pipes/Downsample/view.tsx

```diff
@@ -0,0 +1,114 @@
+import React, {FC, useContext} from 'react'
+import {
+  FlexBox,
+  FlexDirection,
+  ComponentSize,
+  InputLabel,
+  IconFont,
+  ComponentColor,
+  Toggle,
+  InputToggleType,
+  QuestionMarkTooltip,
+  ComponentStatus,
+  MultiSelectDropdown,
+} from '@influxdata/clockface'
+
+import {DURATIONS, FUNCTIONS} from 'src/timeMachine/constants/queryBuilder'
+
+import {PipeContext} from 'src/flows/context/pipe'
+import DurationInput from 'src/shared/components/DurationInput'
+
+import {PipeProp} from 'src/types/flows'
+
+const Downsample: FC<PipeProp> = ({Context}) => {
+  const {data, update} = useContext(PipeContext)
+
+  const selectFn = fn => {
+    const fns = [...data.functions]
+    const fnsFound = fns.map(f => f.name).indexOf(fn)
+    if (fnsFound !== -1) {
```

Comment:
could this be `fns.find(f => f.name == fn)` and then `if (fnsFound) {` ?

Reply:
we need the index for the splice immediately after

Reply:
_le sigh_ ... this is why I shouldn't try and multi-task